packages/block-library/src/icon/style.scss

```diff
@@ -0,0 +1,70 @@
+/**
+ * Editor and frontend styles for the Icon Block.
+ */
+
+/* Icon Block styles. */
+.wp-block-outermost-icon-block {
+	display: flex;
+	line-height: 0;
+
+	// Unset border styling inherited from block props.
+	&.has-border-color {
+		border: none;
+	}
+
+	&.has-icon-color,
+	.has-icon-color {
+		svg {
+			color: currentColor;
+		}
+
+		&:not(.has-no-icon-fill-color) {
+			svg {
+				fill: currentColor;
+			}
+		}
+	}
+
+	// box-sizing is set in the Editor but not on the frontend.
+	.icon-container {
+		box-sizing: border-box;
+	}
+
+	svg {
+		width: 100%;
+		height: 100%;
+		transition: transform 0.1s ease-in-out;
+		transform: rotate(var(--outermost--icon-block--transform-rotate, 0deg)) scaleX(var(--outermost--icon-block--transform-scale-x, 1)) scaleY(var(--outermost--icon-block--transform-scale-y, 1));
+	}
+
+	&.rotate-90,
+	.rotate-90 {
+		--outermost--icon-block--transform-rotate: 90deg;
+	}
+
+	&.rotate-180,
+	.rotate-180 {
+		--outermost--icon-block--transform-rotate: 180deg;
+	}
+
+	&.rotate-270,
+	.rotate-270 {
+		--outermost--icon-block--transform-rotate: 270deg;
+	}
+
+	&.flip-horizontal,
+	.flip-horizontal {
+		--outermost--icon-block--transform-scale-x: -1;
+	}
+
+	&.flip-vertical,
+	.flip-vertical {
+		--outermost--icon-block--transform-scale-y: -1;
+	}
+
+	&.flip-vertical.flip-horizontal,
+	.flip-vertical.flip-horizontal {
+		--outermost--icon-block--transform-scale-x: -1;
+		--outermost--icon-block--transform-scale-y: -1;
+	}
```

Comment:
Remove any CSS related to features that are not supported in the initial implementation.